changelogs/unreleased/build-clang-fails-because-of-null-pointer-subtraction.md

```diff
@@ -0,0 +1,3 @@
+## bugfix/build
+
+* Fix clang errors on debug builds caused by an incorrect NULL pointer subtraction in assertion macros.
```

Comment:
Actually, I don't think this changelog is necessary: this bug could not be observed by users — please drop it.